File: scaf-service/src/main/java/pe/utp/service/CursoService.java
package pe.utp.service;

import org.springframework.data.domain.Pageable;
import pe.utp.dto.PaginateResponseDto;
import pe.utp.dto.curso.CursoRequestDto;
import pe.utp.dto.curso.CursoResponseDto;

public interface CursoService {

    PaginateResponseDto<CursoResponseDto> listar(Pageable pageable);

    CursoResponseDto buscarPorId(Long id);

    CursoResponseDto crear(CursoRequestDto cursoRequestDto);

    CursoResponseDto actualizar(Long id, CursoRequestDto cursoRequestDto);

    void eliminar(Long id);
}

File: scaf-client/src/app/modules/curso/service/curso.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { CursoRequest, CursoResponse } from '../../../core/models/curso.model';
import { PaginateResponse } from '../../../core/models/paginate-response.model';
import { environment } from '../../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class CursoService {
  private readonly apiUrl = `${environment.apiBaseUrl}/api/cursos`;

  constructor(private readonly http: HttpClient) {}

  listar(page: number, size: number): Observable<PaginateResponse<CursoResponse>> {
    return this.http.get<PaginateResponse<CursoResponse>>(this.apiUrl, {
      params: {
        page,
        size
      }
    });
  }

  buscarPorId(id: number): Observable<CursoResponse> {
    return this.http.get<CursoResponse>(`${this.apiUrl}/${id}`);
  }

  crear(curso: CursoRequest): Observable<CursoResponse> {
    return this.http.post<CursoResponse>(this.apiUrl, curso);
  }

  actualizar(id: number, curso: CursoRequest): Observable<CursoResponse> {
    return this.http.put<CursoResponse>(`${this.apiUrl}/${id}`, curso);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
